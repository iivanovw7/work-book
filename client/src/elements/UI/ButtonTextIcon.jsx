import PropTypes from 'prop-types';
import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { ripples } from '../../styles';
import { textColor, textColorLighten } from '../../theme';
/* eslint react/require-default-props: 0 */

const StyledIconButton = styled.button`
	${ripples};
  display: flex;
  flex-wrap: nowrap;
  padding: ${props => props.padding};
  text-align: center;
  flex-direction: column;
  color: ${textColor};
  align-items: center;
  border: none;
  max-width: 48px;
  width: 100%;
  background: none;
  text-decoration: none;
  font-size: ${props => props.fontSize};
  font-style: inherit;
  transition: all 0.2s ease-in-out;
  
  &:hover {
    color: ${textColorLighten};
    transition: all 0.2s ease-in-out;
    user-select: none;
    cursor: pointer;
  }
  
  &:focus {
    outline: none;
  }
  
 
`;

const ButtonTextIcon = (props) => {
  const {
    icon, variant, handleClick, text, theme, padding, fontSize
  } = props;

  return (
    <ThemeProvider theme={{ mode: theme }}>
      <StyledIconButton
        variant={variant}
        onClick={handleClick}
        padding={padding}
        fontSize={fontSize}
      >
        <i className="material-icons">
          {icon}
        </i>
        <span>
          {text}
        </span>
      </StyledIconButton>
    </ThemeProvider>
  );
};

StyledIconButton.propTypes = {
  variant: PropTypes.oneOf(['primary', 'secondary'])
};

StyledIconButton.defaultProps = {
  variant: 'primary'
};

ButtonTextIcon.propTypes = {
  handleClick: PropTypes.func,
  icon: PropTypes.string.isRequired,
  variant: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  fontSize: PropTypes.string.isRequired,
  padding: PropTypes.string.isRequired
};

export default ButtonTextIcon;
