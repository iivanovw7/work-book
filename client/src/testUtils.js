import { createMemoryHistory } from 'history';
import * as localized from './assets/locales.json';

// Mock browser history
export const history = createMemoryHistory('/dashboard');

// Grid config object for testing
export const mappedSizes = {
	xs: 12,
	sm: 12,
	md: 12,
	lg: 11
};

export const localizedText = {
	locales: localized.locales,
	topBar: localized.topBar,
	login: localized.login,
	navigation: localized.navigation,
	pageNotFound: localized.pageNotFound
};
