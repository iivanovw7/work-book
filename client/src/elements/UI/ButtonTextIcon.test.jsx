import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import chalk from 'chalk';
import * as testUtils from '../../testUtils';
import ButtonTextIcon from './ButtonTextIcon';

jest.mock('../../config/apiURL', () => () => '/api');

describe(`Testing [${chalk.yellow('ButtonTextIcon')}] component`, () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    text: 'text',
    theme: 'dark',
    history: testUtils.history,
    message: 'message',
    variant: 'primary',
    icon: 'icon',
    fontSize: '10',
    padding: '12',
    borderRadius: '12',
    direction: 'column'
  };

  const Composition = () => (
    <ButtonTextIcon userAccess {...props} />
  );

  it('Should match snapshot', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition handleClick={mockFunc()} />);

    expect(component).toMatchSnapshot();
  });

  it('Should render with all child elements', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition handleClick={mockFunc()} />);

    expect(component.find('i')).toHaveLength(1);
    expect(component.find('span')).toHaveLength(1);

    expect(component.find('i').text()).toBe('icon');
    expect(component.find('span').text()).toBe('text');
  });

  it('Should render with correct props', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition handleClick={mockFunc()} />);

    expect(component.find('button').prop('fontSize')).toBe('10');
  });

  it('Should trigger button click event', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition handleClick={mockFunc()} />);

    component.find(ButtonTextIcon).at(0).simulate('click');
    expect(component.find(ButtonTextIcon)).toHaveLength(1);
    expect(mockFunc).toHaveBeenCalled();
  });
});
