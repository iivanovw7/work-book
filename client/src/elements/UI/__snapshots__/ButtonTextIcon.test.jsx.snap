// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ButtonTextIcon component Should match snapshot 1`] = `ReactWrapper {}`;
